src/plugins/sailfishoswizards/desktopdocument.cpp

```diff
@@ -0,0 +1,370 @@
+ * \param errorString Output parameter for saving message about possible errors.
+ * \param fileName Path to the associated file.
+ * \param realFileName Name of the auto save that should be loaded if the editor is opened from an auto save file.
+ * It's the same as \a fileName if the editor is opened from a regular file.
+ * \sa OpenResult
+ */
+IDocument::OpenResult DesktopDocument::open(QString *errorString, const QString &fileName,
+                                            const QString &realFileName)
+{
+    Q_UNUSED(errorString);
+    Q_UNUSED(realFileName);
+    QFile file(fileName);
+    if (file.open(QFile::ReadOnly)) {
+        this->m_fileName = file.fileName();
+        this->m_applicationPath = QFileInfo(file).dir().path();
+        setFilePath(Utils::FileName::fromString(fileName));
+
+        this->m_editorWidget->setContent(file.readAll(), this->m_applicationPath, this->m_resolutions);
+        m_oldIcons = m_editorWidget->iconPaths();
+        m_iconsIndex = QFileInfo(
+                           m_oldIcons.value("172x172")).fileName().split("_").last().split(".").first().toInt();
+        setModified(false);
+        return OpenResult::Success;
+    }
+    return OpenResult::ReadError;
+}
+
+/*!
+ * \brief Reopens the file that is already opened.
+ * Returns \c true if the file was opened and read successfully, otherwise returns \c false.
+ * \param errorString Output parameter for saving message about possible errors.
+ */
+bool DesktopDocument::reload(QString *errorString, ReloadFlag flag, ChangeType type)
+{
+    Q_UNUSED(flag);
+    Q_UNUSED(type);
+    m_editorWidget->clearTranslations();
+    if (open(errorString, this->m_fileName, this->m_fileName) == OpenResult::Success)
+        return true;
+    else
+        return false;
+}
+
+/*!
+ * \brief This slot sets the modified status to \c true, when the associated signal is emitted.
+ * The slot wraps the proper \c setModified() method, because of its non-parameter prototype, which is much more comfortable to use when connecting signals.
+ * \sa setModified()
+ */
+void DesktopDocument::requestSave()
+{
+    setModified();
+}
+
+/*!
+ * \brief Counts the unoccupied index for naming and adding of new set of icons to configuration files and target project directory.
+ * \sa addIconFiles(), writeIconsToPro(), writeIconsToYaml(), writeIconsToSpec()
+ */
+void DesktopDocument::countIconIndex()
+{
+    QDir projectDir(m_applicationPath);
+    QRegExp iconsPrefixCheck(SailfishOSWizards::Constants::REG_EXP_FOR_ICONS_PROFILE_PREFIX);
+
+    const QString proFilePath = Common::getProFilePath(m_applicationPath);
+    if (!proFilePath.isEmpty()) {
+        QFile profile(proFilePath);
+        if (profile.open(QIODevice::ReadOnly)) {
+            m_iconsIndex = 1;
+            QStringList  strListPro;
+            while (!profile.atEnd())
+                strListPro << QTextCodec::codecForName("UTF-8")->toUnicode(profile.readLine());
```

Comment:
Isn't this identical to "strListPro << QString::fromUtf8(profile.readLine())" ?

Reply:
Fixed